flexget/plugins/input/npo_watchlist.py

```diff
@@ -95,173 +105,136 @@ def _login(self, task, config):
 
             if 'isAuthenticatedUser' not in profile_response.cookies:
                 raise plugin.PluginError('Failed to login. Check username and password.')
+
         except RequestException as e:
             raise plugin.PluginError('Request error: %s' % e.args[0])
 
-    def _get_page(self, task, config, url):
-        self._login(task, config)
+    def _get_favourites_entries(self, task, config, profileId):
+        # Details of profile using the $profileId:
+        # https://www.npostart.nl/api/account/@me/profile/$profileId  (?refresh=1)
+        # XMLHttpRequest could also be done on, but is not reliable, because if a series is
+        # bookmarked after episode is already broadcasted, it does not appear in this list.
+        # https://www.npostart.nl/ums/accounts/@me/favourites/episodes?dateFrom=2018-06-04&dateTo=2018-06-05
+        profile_details_url = 'https://www.npostart.nl/api/account/@me/profile/'
 
+        entries = []
         try:
-            page_response = requests.get(url)
-            if page_response.url != url:
-                raise plugin.PluginError('Unexpected page: {} (expected {})'.format(page_response.url, url))
-
-            return page_response
+            profile = requests.get(profile_details_url + profileId).json()
+            log.info('Retrieving favorites for profile %s', profile['name'])
+            for favourite in profile['favourites']:
+                if favourite['mediaType'] == 'series':
+                    entries += self._get_series_episodes(task, config, favourite['mediaId'])
         except RequestException as e:
             raise plugin.PluginError('Request error: %s' % e.args[0])
+        return entries
 
-    def _get_watchlist_entries(self, task, config, page):
-        watchlist = page.find('div', attrs={'id': 'component-grid-watchlist'})
-        return self._parse_tiles(task, config, watchlist)
-
-    _series_info = dict()
-
-    def _get_series_info(self, task, config, episode_name, episode_url):
-        log.info('Retrieving series info for %s', episode_name)
-        try:
-            response = requests.get(episode_url)
-            page = get_soup(response.content)
+    def _get_series_episodes(self, task, config, mediaId, series_info=None, page=1):
+        episode_tiles_url = 'https://www.npostart.nl/media/series/{0}/episodes?page={1}'
+        episode_tiles_parameters = '&tilemapping=dedicated&tiletype=asset'
```

Comment:
i think it's better to hold these as a dict and pass it via `params` keyword to request